src/controllers/midi/midicontroller.h

```diff
@@ -1,12 +1,28 @@
 #pragma once
 
+#include <QJSValue>
+#include <utility>
+
 #include "controllers/controller.h"
 #include "controllers/midi/legacymidicontrollermappingfilehandler.h"
 #include "controllers/midi/midimessage.h"
 #include "controllers/softtakeover.h"
 
 class MidiOutputHandler;
 
+class MidiInputHandleJSProxy final : public QObject {
+    Q_OBJECT
+  public:
+    MidiInputHandleJSProxy(
+            const std::shared_ptr<LegacyMidiControllerMapping> mapping,
+            const MidiInputMapping& inputMapping);
+    Q_INVOKABLE bool disconnect();
+
+  protected:
+    std::shared_ptr<LegacyMidiControllerMapping> m_mapping;
+    MidiInputMapping m_inputMapping;
```

Comment:
Is there a risk of memory leak here? I'm not sure whether or not a `MidiInputHandleJSProxy` will ever be collected and in which conditions.